src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -101,31 +101,27 @@
 typedef id (^SyncWorkQueueBlockWithReturnValue)(void);
 typedef BOOL (^SyncWorkQueueBlockWithBoolReturnValue)(void);
 
-@interface MTRDeviceController () {
+@implementation MTRDeviceController {
     // Atomic because it can be touched from multiple threads.
     std::atomic<chip::FabricIndex> _storedFabricIndex;
-}
-
-// queue used to serialize all work performed by the MTRDeviceController
-@property (atomic, readonly) dispatch_queue_t chipWorkQueue;
 
-@property (readonly) chip::Controller::DeviceCommissioner * cppCommissioner;
```

Comment:
This one did not say `nonatomic` so presumably was atomic?  How does the change to ivar work there?

Reply:
`_commissionableBrowser` is only accessed from the work queue.
`_nodeIDToDeviceMap` is protected by a dedicated lock.
`_cppCommissioner` is generally only accessed from the work queue (because we can only make calls into it from the work queue), apart from the `isRunning` check.
All other properties where only accessed directly via the ivar already in all cases.

Reply:
That leaves `isRunning`... I think given that the write access `_cppCommissioner = nullptr;` in shutDown wasn't using the setter means that going through the getter to read will have had very limited (if any) benefit there. We mostly call `isRunning` from the chip queue, which is fine, and the remainder seem to be these "check first before we dispatch" cases, that we need to address as part of #31145.

Reply:
Sounds good.